openapi-v1.yaml

```diff
@@ -2682,6 +2692,118 @@ definitions:
+        # Semantically this is equivalent to `idle`.
+      - succeeded
+        # The client has declared that the task graph has completed
+        # successfully. This ordinarily entails all nodes succeeding.
+      - failed
+        # The client has decided that the task graph was not successful.
+      - cancelled
+        # The client has cancelled further execution of the task graph.
+
+  TaskGraphLog:
+    description: Logging information about the execution of a task graph.
+    type: object
+    properties:
+      uuid:
+        type: string
+        readOnly: true
+        description: The server-generated UUID of the task graph.
+      namespace:
+        type: string
+        readOnly: true
+        description: The namespace that owns this task graph log.
+      created_by:
+        type: string
+        readOnly: true
+        description: The name of the user who created this task graph log.
+      start_time:
+        type: string
+        format: date-time
+        readOnly: true
+        x-omitempty: true
+        description: >
+          The start time of the task graph, recorded when the server
+          starts executing the first node.
+      end_time:
+        type: string
+        format: date-time
+        readOnly: true
+        x-omitempty: true
+        description: >
+          The end time of the task graph, recorded when the client reports
+          completion.
+      status:
+        $ref: '#/definitions/TaskGraphLogStatus'
+      nodes:
+        type: array
+        items:
+          $ref: '#/definitions/TaskGraphNodeMeta'
+        description: >
+          The structure of the graph. This is provided by the client
+          when first setting up the task graph. Thereafter, it is read-only.
+
+  TaskGraphNodeMeta:
+    description: Metadata about an individual node in a task graph.
+    type: object
+    properties:
+      client_node_uuid:
+        type: string
+        description: The client-generated UUID of the given graph node.
+      name:
+        type: string
+        description: >
+          The client-generated name of the node.
+          This is not guaranteed to be unique.
+      depends_on:
+        type: array
+        items:
+          type: string
+        description: >
+          The client_node_uuid of each node that this node depends upon.
+          Used to define the structure of the graph.
+      executions:
+        type: array
+        items:
+          $ref: '#/definitions/ArrayTask'
+        readOnly: true
+        description: >
+          ArrayTasks representing each execution attempt for this node.
+          For nodes that have never been submitted, this will be empty.
+          For nodes that have been retried, this may have multiple entries.
+          The last one in the list represents the most recent execution.
+
+          This is read-only and generated by the server based on
+          the tasks it has actually executed.
+
+  TaskGraphLogsData:
```

Comment:
Naming of responses is difficult. We use `Data`, `Response`, `Page` I think the latter is more appropriate here

Reply:
I only see `[DataType]Data` in this file (`ArrayTaskData`, `ArrayBrowserData`, `ArrayEndTimestampData`, etc.), so I chose this to be consistent with all of them.